### PR TITLE
[WFLY-10110][WFLY-10111] Fix attribute setting for JCA pool configuration initial-size and blocking-timeout-wait-milis

### PR DESCRIPTION
**Pull request by fl4via**
Jiras:
https://issues.jboss.org/browse/WFLY-10111
https://issues.jboss.org/browse/WFLY-10110

### CODE REVIEW

**Comment by maeste** [comment]
@fl4via failing test seems related and probably need to be fixed (test itself). Otherwise LGTM

**Comment by fl4via** [comment]
Retest this please

**Review by kabir — COMMENTED** [review]
@fl4via I believe you need a model version bump in the resource-adapters subsystem. DataSources seems to already be bumped for WF13. I don't think this is used by JCA, but if it is we need to bump the version there as well I think.

Since you are changing the default value, you probably need to revisit the transformers too?

File: connector/src/main/java/org/jboss/as/connector/subsystems/common/pool/PoolConfigurationRWHandler.java [hunk 1]
```diff
@@ -137,10 +137,10 @@ private void updatePoolConfigs(List<PoolConfiguration> poolConfigs, String param
                     pc.setMinSize(newValue.asInt());
                 }
                 if (INITIAL_POOL_SIZE.getName().equals(parameterName)) {
-                    pc.setInitialSize(newValue.asInt());
+                    pc.setInitialSize(newValue.isDefined()? newValue.asInt(): 0);
```
Comment (kabir) [inline connector/src/main/java/org/jboss/as/connector/subsystems/common/pool/PoolConfigurationRWHandler.java]: To me this looks like the default value is taken to be 0?

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/poolattributes/ResourceAdapterPoolAttributesTestCase.java [hunk 2]
```diff
@@ -152,7 +152,7 @@ public void testModifyPoolAttributes() throws Exception {
         Assert.assertNotNull(poolConfiguration);
         Assert.assertEquals(2, poolConfiguration.getMinSize());
         Assert.assertEquals(5, poolConfiguration.getMaxSize());
-        Assert.assertEquals(0, poolConfiguration.getInitialSize());
+        Assert.assertEquals(2, poolConfiguration.getInitialSize());
```
Comment (kabir) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/poolattributes/ResourceAdapterPoolAttributesTestCase.java]: Where does this 2 come from?

**Comment by maeste** [comment]
@kabir opened https://github.com/wildfly/wildfly/pull/11154 with required fixes.
You can close this one.

**Pull request closed without merging**